mlx/traceability.py

```diff
@@ -249,6 +253,38 @@ def run(self):
 
         return [item_list_node]
 
+class ItemLinkDirective(Directive):
+    """
+    Directive to generate a list of items.
+
+    Syntax::
+
+      .. item-link::
+         :sources: list_of_items
+         :targets: list_of_items
+         :type: relationship_type
+
+    """
+    final_argument_whitespace = True
+    # Options
+    option_spec = {'sources': directives.unchanged,
+                   'targets': directives.unchanged,
+                   'type': directives.unchanged}
+    # Content disallowed
+    has_content = False
+
+    def run(self):
+        env = self.state.document.settings.env
+        app = env.app
+
+        item_link_node = ItemLink('')
+
+        item_link_node['sources'] = self.options['sources']
+        item_link_node['targets'] = self.options['targets']
+        item_link_node['type'] = self.options['type']
```

Comment:
if one of these arguemtns to item-link is not given, the tool will crash on the access of the options['...'].

2 solutions:
- can we make the arguments mandatory: sphinx checks and reports if the argument is missing (preferred)
- we probably can't live without these arguments, so use `report_warning` in order to inform the user